### PR TITLE
handle null images

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes RNBW-2871

## What changed (plus any additional context for devs)
we handle undefined images fine for out image handling but mmkv doesn't like undefined keys so I've added a default case 

## PoW (screenshots / screen recordings)
https://cloud.skylarbarrera.com/Screen-Recording-2022-03-20-12-37-48.mp4

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

File: src/hooks/usePersistentAspectRatio.ts [hunk 1]
```diff
@@ -23,26 +23,31 @@ type Result = {
...
       setState(State.loading);
-      Image.getSize(
-        lowResUrl,
-        (width, height) => {
-          setAspectRatio(width / height);
-          setState(State.loaded);
-        },
-        () => setState(State.failed)
-      );
+      lowResUrl !== null &&
```
Comment (osdnk) [inline src/hooks/usePersistentAspectRatio.ts:37]: when `lowResUrl` is null?

File: src/hooks/usePersistentAspectRatio.ts [hunk 2]
```diff
@@ -23,26 +23,31 @@ type Result = {
 };
 
 export default function usePersistentAspectRatio(url: string): Result {
+  if (url === null) url = 'default';
```
Comment (osdnk) [inline src/hooks/usePersistentAspectRatio.ts:26]: not a fan of magic words and overriding arguments. This is dirty and I don't want to establish this as an acceptable pattern

File: src/hooks/usePersistentAspectRatio.ts (hunk 1 above)
Comment (skylarbarrera) [reply src/hooks/usePersistentAspectRatio.ts:37]: this I will remove

**Review by osdnk — CHANGES_REQUESTED** [review]
Won't this be simpler to change condition in `useEffect` to `if (state === State.init && nonSvgUrl && url)`

and then:

`useMMKVNumber((nonSvgUrl || '') as string, storage);`

File: src/hooks/usePersistentAspectRatio.ts (hunk 1 above)
Comment (osdnk) [reply src/hooks/usePersistentAspectRatio.ts:37]: I think this can be `undefined` tho. If signing will throw an exception. see `src/handlers/imgix.ts:91`

**Comment by osdnk** [comment]
or maybe we can write a hook `useMMKVNumberWithFallback` that will wrap this exception and return always a default value (eg., ratio 1):

`const [ratio, setAspectRatio] = useMMKVNumberWithFallback(nonSvgUrl as string, storage), 1)`

**Comment by skylarbarrera** [comment]
> Won't this be simpler to change condition in `useEffect` to `if (state === State.init && nonSvgUrl && url)`
> 
> and then:
> 
> `useMMKVNumber((nonSvgUrl || '') as string, storage);`
> 
> ?

sure thing

**Comment by skylarbarrera** [comment]
> or maybe we can write a hook `useMMKVNumberWithFallback` that will wrap this special case and return always a default value then (eg., ratio 1):
> 
> `const [ratio, setAspectRatio] = useMMKVNumberWithFallback(nonSvgUrl as string, storage), 1)`

if we want to do this later be my guest but I just want to fix the crash and get a release out

**Comment by osdnk** [comment]
so please fix both files and remove commented out code

File: src/hooks/usePersistentAspectRatio.ts [hunk 3]
```diff
@@ -23,14 +23,18 @@ type Result = {
...
```
Comment (osdnk) [inline src/hooks/usePersistentAspectRatio.ts:26]:
```suggestion
```

**Review by osdnk — APPROVED** [review]

**Review by tchayen — APPROVED** [review]

**Pull request merged**